File: CraftyEditor/Class/Ogame/Ogame.cs
using System;
using OpenTK;

namespace CraftyEditor
{
    public class Ogame : Singleton<Ogame>
    {
        public bool initialize, running;

        public Ogame() : base()
        {
            initialize = true;
            running = true;
        }

        public void Run()
        {
            while(running)
            {
                if (initialize)
                {
                    Initialize();
                    initialize = false;
                }
                Clear();
                Update();
                Draw();
            }
        }

        public void Close() { running = false; }

        public virtual void Initialize()
        {
            //...
        }

        public virtual void Clear()
        {
            //...
        }

        public virtual void Update()
        {
            //...
        }

        public virtual void Draw()
        {
            //...
        }
    }
}

File: CraftyEditor/OGAME/Ogame.cpp
#pragma once
#include "Controller/GraphicController.cpp"

class Ogame
{
	public: GraphicController graphics;

	public: Ogame() {}

	public: void run(string caption, int width, int height)
	{
		graphics.initialize(caption, width, height);
		Load();

		while (graphics.isOpen())
		{
			while (graphics.hasEvents())
			{
				if (graphics.hasClosed()) graphics.close();

				//...
			}

			graphics.clear("#adaaa1");

			Update();
			Draw();
			
			graphics.swap();
		}
	}

	public: virtual void Load() { graphics.initialized(); }

	public: virtual void Update() { }

	public: virtual void Draw() { }
};